pkg/monitortests/network/onpremhaproxy/monitortest.go

```diff
@@ -0,0 +1,291 @@
+package onpremhaproxy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortestlibrary/podaccess"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type operatorLogAnalyzer struct {
+	kubeClient kubernetes.Interface
+}
+
+func InitialAndFinalOperatorLogScraper() monitortestframework.MonitorTest {
+	return &operatorLogAnalyzer{}
+}
+
+func (w *operatorLogAnalyzer) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	var err error
+	w.kubeClient, err = kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := scanAllOperatorPods(ctx, w.kubeClient, newOperatorLogHandler(recorder)); err != nil {
+		return fmt.Errorf("unable to scan operator logs: %w", err)
+	}
+
+	return nil
+}
+
+func scanAllOperatorPods(ctx context.Context, kubeClient kubernetes.Interface, logHandlers ...podaccess.LogHandler) error {
+	pods, err := kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("couldn't list pods: %w", err)
+	}
+
+	errs := []error{}
+	for _, pod := range pods.Items {
+		if !(strings.HasPrefix(pod.Namespace, "openshift-kni-infra") || strings.HasPrefix(pod.Namespace, "openshift-openstack-infra") || strings.HasPrefix(pod.Namespace, "openshift-vsphere-infra")) {
```

Comment:
Lets list pods from these namespaces instead of listing all pods?

Reply:
Done